src/v/cluster/self_test_rpc_types.h

```diff
@@ -341,15 +341,19 @@ struct empty_request
 struct start_test_request
   : serde::envelope<
       start_test_request,
-      serde::version<1>,
+      serde::version<2>,
       serde::compat_version<0>> {
     using rpc_adl_exempt = std::true_type;
 
     uuid_t id;
     std::vector<diskcheck_opts> dtos;
     std::vector<netcheck_opts> ntos;
-    std::vector<cloudcheck_opts> ctos;
     std::vector<unknown_check> unknown_checks;
+    std::vector<cloudcheck_opts> ctos;
+
+    auto serde_fields() {
+        return std::tie(id, dtos, ntos, unknown_checks, ctos);
+    }
```

Comment:
@WillemKauf i'm confused about what is going on here, and i can't figure out the answer from the commit message.

looking at the git history i see when version was 0, the order of serde fields in the struct was d,n,c. then it looks like it was bumped to 1 when the unknown checks were appended, which seems correct.

but then here, the commit appears to claim that a bug is being fixed (i'm not sure what the bug was), and also, the ordering of the fields was changed and that's not allowed even when bumping the version.